server.go

```diff
@@ -1,10 +1,42 @@
 package dnscrypt
 
 import (
+	"net"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
```

Comment:
I'm not sure what the style is on the project, of if there even is one, but in some files the external packages are all clumped together, while in this file they are grouped by hostname.